test/solidity/Facets/EcoFacet.t.sol

```diff
@@ -0,0 +1,492 @@
+            encodedRoute: borshEncodedRoute
+        });
+
+        // Approve USDC
+        usdc.approve(
+            _facetTestContractAddress,
+            bridgeData.minAmount + TOKEN_SOLVER_REWARD
+        );
+
+        // Expect events
+        vm.expectEmit(true, true, true, true, _facetTestContractAddress);
+        emit BridgeToNonEVMChain(
+            bridgeData.transactionId,
+            bridgeData.destinationChainId,
+            solanaAddress
+        );
+
+        vm.expectEmit(true, true, true, true, _facetTestContractAddress);
+        emit LiFiTransferStarted(bridgeData);
+
+        // Execute bridge
+        ecoFacet.startBridgeTokensViaEco(bridgeData, ecoData);
+
+        vm.stopPrank();
+    }
+
+    function testBridge_ToTron() public {
+        vm.startPrank(USER_SENDER);
+
+        // Set up bridge data for Tron
+        bridgeData.destinationChainId = LIFI_CHAIN_ID_TRON;
+        bridgeData.receiver = USER_RECEIVER; // Can use regular address for Tron
+
+        // Create mock encoded route data
+        bytes memory mockEncodedRoute = hex"0102030405060708090a0b0c0d0e0f10";
+
+        EcoFacet.EcoData memory ecoData = EcoFacet.EcoData({
+            receiverAddress: USER_RECEIVER,
+            nonEVMReceiver: "",
+            prover: address(0x1234),
+            rewardDeadline: uint64(block.timestamp + 2 days),
+            solverReward: TOKEN_SOLVER_REWARD,
+            encodedRoute: mockEncodedRoute // Now required for Tron too
+        });
+
+        // Approve USDC
+        usdc.approve(
+            _facetTestContractAddress,
+            bridgeData.minAmount + TOKEN_SOLVER_REWARD
+        );
+
+        // Expect event
+        vm.expectEmit(true, true, true, true, _facetTestContractAddress);
+        emit LiFiTransferStarted(bridgeData);
+
+        // Execute bridge
+        ecoFacet.startBridgeTokensViaEco(bridgeData, ecoData);
+
+        vm.stopPrank();
+    }
+
+    function testRevert_WithoutEncodedRoute() public {
+        vm.startPrank(USER_SENDER);
+
+        // Test with any destination chain
+        bridgeData.destinationChainId = 10; // Optimism
+
+        // Create EcoData without encodedRoute
+        EcoFacet.EcoData memory ecoData = EcoFacet.EcoData({
+            receiverAddress: USER_RECEIVER,
+            nonEVMReceiver: "",
+            prover: address(0x1234),
+            rewardDeadline: uint64(block.timestamp + 2 days),
+            solverReward: TOKEN_SOLVER_REWARD,
+            encodedRoute: "" // Missing encodedRoute (now required for all chains)
+        });
+
+        usdc.approve(
+            _facetTestContractAddress,
+            bridgeData.minAmount + TOKEN_SOLVER_REWARD
+        );
+
+        vm.expectRevert(InvalidConfig.selector);
+        ecoFacet.startBridgeTokensViaEco(bridgeData, ecoData);
+
+        vm.stopPrank();
+    }
+
+    // Override base test since we no longer support destination calls
+    function testBase_Revert_BridgeWithInvalidDestinationCallFlag()
```

Comment:
This test is problematic. 

1) it has revert in its name but it doesnt revert
2) we seem to have no tests where this flag is set to true
3) the comments and the test setup seem incoherent to me